### PR TITLE
feat: add grant role resource

### PR DESCRIPTION
**Pull request by sfc-gh-swinkler**
adds snowflake_grant_role resource. Should this be renamed to snowflake_grant_account_role instead?
https://docs.snowflake.com/en/sql-reference/sql/grant-role

### CODE REVIEW

File: pkg/resources/grant_role_acceptance_test.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+			"role_name":        config.StringVariable(roleName),
+			"parent_role_name": config.StringVariable(parentRoleName),
+		}
+	}
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
+		PreCheck:                 func() { acc.TestAccPreCheck(t) },
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.RequireAbove(tfversion.Version1_5_0),
+		},
```
Comment (sfc-gh-jcieslak) [inline pkg/resources/grant_role_acceptance_test.go:31]: There's no check for delete

**Review by sfc-gh-jcieslak — CHANGES_REQUESTED** [review]

**Comment by sfc-gh-asawicki** [comment]
> adds snowflake_grant_role resource. Should this be renamed to snowflake_grant_account_role instead? https://docs.snowflake.com/en/sql-reference/sql/grant-role

I think this is a valid idea. We have *_role for every other planned resource. @sfc-gh-jcieslak wdyt?

File: pkg/resources/grant_role.go [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+				if err := d.Set("role_name", parts[0]); err != nil {
+					return nil, err
+				}
+				switch parts[1] {
+				case "ROLE":
+					if err := d.Set("parent_role_name", parts[2]); err != nil {
+						return nil, err
+					}
+				case "USER":
+					if err := d.Set("usere_name", parts[2]); err != nil {
```
Comment (sfc-gh-asawicki) [inline pkg/resources/grant_role.go:68]: usere_name -> user_name

Comment (sfc-gh-asawicki) [reply pkg/resources/grant_role.go:68]: It was not spotted by tests, so let's add an import step to both of the tests pls.

File: pkg/resources/grant_role.go [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+		d.SetId("")
+	}
+
+	return nil
+}
+
+func DeleteGrantRole(d *schema.ResourceData, meta interface{}) error {
+	db := meta.(*sql.DB)
+	client := sdk.NewClientFromDB(db)
+	parts := strings.Split(d.Id(), helpers.IDDelimiter)
```
Comment (sfc-gh-asawicki) [inline pkg/resources/grant_role.go:155]: we could have a safety check like in the importer func here (i.e. `if len(parts) != 3)`). But this is an optional addition.

**Review by sfc-gh-asawicki — CHANGES_REQUESTED** [review]

File: examples/resources/snowflake_grant_role/import.sh [hunk 4]
```diff
@@ -0,0 +1,2 @@
+# format is role_name (string) | grantee_object_type (ROLE|USER) | grantee_name (string)
```
Comment (sfc-gh-asawicki) [inline examples/resources/snowflake_grant_role/import.sh:1]: Hmm, the docs were not generated for this new resource (the check did not fail btw, we should check why).

**Review by sfc-gh-asawicki — CHANGES_REQUESTED** [review]

File: examples/resources/snowflake_grant_role/import.sh (hunk 4 above)
Comment (sfc-gh-swinkler) [reply examples/resources/snowflake_grant_role/import.sh]: ran `make docs`

File: pkg/resources/grant_role.go (hunk 3 above)
Comment (sfc-gh-swinkler) [reply pkg/resources/grant_role.go]: added check

File: pkg/resources/grant_role_acceptance_test.go (hunk 1 above)
Comment (sfc-gh-swinkler) [reply pkg/resources/grant_role_acceptance_test.go]: added check destroy func

**Review by sfc-gh-jcieslak — DISMISSED** [review]

**Review by sfc-gh-asawicki — DISMISSED** [review]

**Pull request merged**